text/0000-user-ids.md

```diff
@@ -0,0 +1,42 @@
+# Secure runtime environments
+
+## Summary
+
+This RFC recommends that Paketo Buildpacks should in general strive to avoid making run-time modifications to the Buildpacks layers or the application directory by default. Instead, they should consider the output image to be read-only by default and explicitly set the parts that need to be writeable during runtime.
+
+
+## Motivation
+
+As of Buildpacks [RFC 0085](https://github.com/buildpacks/rfcs/blob/main/text/0085-run-uid.md) the Cloud Native Buildpacks project recommends that the run-time and build-time user ids be different. The motivation behind this RFC was to improve the security of output images and prevent arbitrary runtime modifications to built artifacts.
+
+The run-time and build-time user can potentially share the same group id if they wish to make modifications to the output image during run-time, by changing the permissions of the files they wish to modify during run-time to be group writable during the build process. Ideally most buildpacks should use the temporary directories or other known scratch directories for ephemeral runtime files.
```

Comment:
Another reason to do this: in some hardened k8s environments the PodSecurityPolicy requires that the use of a read only root file system. In these environments a writable tmpfs is typically mounted to necessary locations like `/tmp` and `/run`.

However, in the above case we cannot use group level permissions as a workaround.